modules/pricer/src/main/java/com/opengamma/strata/pricer/calibration/MarketDataFxRateProvider.java

```diff
@@ -33,7 +35,19 @@ public double fxRate(Currency baseCurrency, Currency counterCurrency) {
     if (baseCurrency.equals(counterCurrency)) {
       return 1;
     }
-    FxRate rate = marketData.getValue(FxRateKey.of(baseCurrency, counterCurrency));
-    return rate.fxRate(baseCurrency, counterCurrency);
+    // Try direct pair
```

Comment:
Could this logic be implemented in a decorator that implements `FxRateProvider` and delegates to another `FxRateProvider`? Then it would be more universally useful.